doc/manual/arrays.rst

```diff
@@ -213,6 +213,37 @@ that the result is of type ``Float64`` by writing::
 
     Float64[ 0.25*x[i-1] + 0.5*x[i] + 0.25*x[i+1] for i=2:length(x)-1 ]
 
+.. _man-generator-expressions:
+
+Generator Expressions
+---------------------
+
+Comprehensions can also be written without the enclosing square brackets, producing
+an object known as a generator. This object can be iterated to produce values on
+demand, instead of allocating an array and storing them in advance
+(see :ref:`_man-interfaces-iteration`).
+For example, the following expression sums a series without allocating memory::
+
+    julia> sum(1/n^2 for n=1:1000)
+    1.6439345666815615
+
+When writing a generator expression with multiple dimensions, it needs to be
+enclosed in parentheses to avoid ambiguity::
+
+    julia> collect(1/(i+j) for i=1:2, j=1:2)
+    ERROR: function collect does not accept keyword arguments
```

Comment:
I'm in favor of after a `for i=1:2` in parens having both `, j=1:2` and `, j in 1:2` be interpreted as more of generator. If not that, then an error. Another note is that if we eliminated the comma from the comprehension syntax, then I believe this issue would really go away:

```jl
[1/(i+j) for i=1:2 j=1:2]
collect(1/(i+j) for i=1:2 j=1:2)
```

This is currently a syntax error.